server/chat-commands/info.ts

```diff
@@ -2535,35 +2535,41 @@ export const commands: ChatCommands = {
 		if (!room.settings.quotes?.length) return this.errorReply(`This room has no quotes.`);
 		this.runBroadcast();
 		const {quote, date, userid} = room.settings.quotes[Math.floor(Math.random() * room.settings.quotes.length)];
-		const formatted = quote.split('\n').map(item => Chat.formatText(item)).join('<br />');
 		const time = Chat.toTimestamp(new Date(date), {human: true});
-		const attribution = toID(target) === 'showauthor' ? `<hr /><small>Added by ${userid} on ${time}</small>` : '';
-		return this.sendReplyBox(`${formatted}<br />` + attribution);
+		const attribution = toID(target) === 'showauthor' ? `<br /><hr /><small>Added by ${userid} on ${time}</small>` : '';
+		return this.sendReplyBox(`${Chat.formatText(quote).replace(/\n/g, '<br />')}${attribution}`);
 	},
 	randquotehelp: [`/randquote [showauthor] - Show a random quote from the room. Add 'showauthor' to see who added it and when.`],
 
 	addquote: 'quote',
 	quote(target, room, user) {
 		room = this.requireRoom();
+		if (!room.persist) {
+			return this.errorReply("This command is unavailable in temporary rooms.");
+		}
 		target = target.trim();
 		this.checkCan('mute', null, room);
+		if (!target) {
+			return this.parse(`/help quote`);
+		}
 		if (!room.settings.quotes) room.settings.quotes = [];
 		if (this.filter(target) !== target) {
 			return this.errorReply(`Invalid quote.`);
 		}
 		if (room.settings.quotes.filter(item => item.quote === target).length) {
 			return this.errorReply(`"${target}" is already quoted in this room.`);
 		}
-		if (target.length > 300) {
-			return this.errorReply(`Your quote is too long.`);
+		if (target.length > 8192) {
+			return this.errorReply(`Your quote cannot exceed 8192 characters.`);
 		}
 		if (room.settings.quotes.length >= 100) {
 			return this.errorReply(`This room already has 100 quotes, which is the maximum.`);
 		}
 		room.settings.quotes.push({userid: user.id, quote: target, date: Date.now()});
 		room.saveSettings();
-		this.privateModAction(`${user.name} added the quote "${target}"`);
-		return this.modlog(`ADDQUOTE`, null, target);
+		const collapsedQuote = Chat.collapseLineBreaksHTML(target).replace(/&#10;/g, ' ');
```

Comment:
Oh, wait, I think I understand now. You're just abusing `collapseLineBreaksHTML`. But why? Wouldn't `target.replace(/\n/g, ' ')` be shorter and also less buggy?

Reply:
yes, idk why i didnt think i could do that